src/node_crypto.cc

```diff
@@ -5616,13 +5616,13 @@ void RandomBytesProcessSync(Environment* env,
 void RandomBytes(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
-  if (!args[0]->IsUint32()) {
+  if (!args[0]->IsNumber() || args[0]->NumberValue() < 0) {
     return env->ThrowTypeError("size must be a number >= 0");
   }
 
   const int64_t size = args[0]->IntegerValue();
-  if (size < 0 || size > Buffer::kMaxLength)
-    return env->ThrowRangeError("size is not a valid Smi");
+  if (size > Buffer::kMaxLength)
+    return env->ThrowTypeError("size must be a uint32");
```

Comment:
I'm +1 on this change, but wouldn't feel comfortable with it *not* being semver-major since it changes an error message

Reply:
Wait, why is the RangeError changed to a TypeError?

Reply:
Good question: A range error made no sense there - and the code path emitting it didn't happen in reasonable cases anyway. For example if you passed -1.1 you'd get the TypeError in the path above - so not changing this to a TypeError would break a lot more.

Reply:
Hmm, I am not sure I totally buy the argument of breaking less code here. Since we are changing the error messages here, why not make everything Semantically Correct(r)?

- -1.1: **RangeError**: size must be a uint32
- 2<sup>40</sup>: **RangeError**: size must be a uint32
- 'hello': **TypeError**: size must be a number

Reply:
I'm not saying that a `TypeError` is more correct here, I'll try to explain what I mean by breakage:

Before both `-1.1` _and_ `2<sup>40</sup>` were `TypeError`s since `->IsUint32` returns false and this change would make them both `RangeError`s. 

A `RangeError` would _only_ happen if the number was a `Uint32` _and_ it's smaller than `Buffer::kMaxLength` which would only happen if the number was bigger than `2147483647` which is never. Only on 32 bit architecture buffers are max sized at `1073741823` which means RangeErrors would _only_ happen on 32 bit and only when the number passed was between 1073741823 and 2147483647 - which I guess is very rare.

I'm fine with breaking this in a semver-major, but I'd rather do that in a different PR titled "make invalid numbers throw `RangeError` in randomBytes so it can be specifically examined for that change.

Reply:
That's fair :+1: